src/Symfony/Component/Notifier/Notification/Notification.php

```diff
@@ -0,0 +1,195 @@
+        $notification->exception = $exception;
+
+        return $notification;
+    }
+
+    /**
+     * @return $this
+     */
+    public function subject(string $subject)
+    {
+        $this->subject = $subject;
+
+        return $this;
+    }
+
+    public function getSubject(): string
+    {
+        return $this->subject;
+    }
+
+    /**
+     * @return $this
+     */
+    public function content(string $content)
+    {
+        $this->content = $content;
+
+        return $this;
+    }
+
+    public function getContent(): string
+    {
+        return $this->content;
+    }
+
+    /**
+     * @return $this
+     */
+    public function importance(string $importance)
+    {
+        $this->importance = $importance;
+
+        return $this;
+    }
+
+    public function getImportance(): string
+    {
+        return $this->importance;
+    }
+
+    /**
+     * @param string $level A PSR Logger log level name
+     *
+     * @return $this
+     */
+    public function importanceFromLogLevelName(string $level)
+    {
+        $level = self::$levels[strtolower($level)];
+        $this->importance = $level >= 500 ? self::IMPORTANCE_URGENT : ($level >= 400 ? self::IMPORTANCE_HIGH : self::IMPORTANCE_LOW);
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function emoji(string $emoji)
+    {
+        $this->emoji = $emoji;
+
+        return $this;
+    }
+
+    public function getEmoji(): string
+    {
+        if (!$this->emoji && $this->exception) {
+            switch ($this->importance) {
+                case self::IMPORTANCE_URGENT:
+                    return '🌩️';
+                case self::IMPORTANCE_HIGH:
+                    return '🌧️';
+                case self::IMPORTANCE_MEDIUM:
+                    return '🌦️';
+                case self::IMPORTANCE_LOW:
+                default:
+                    return '⛅';
+            }
+        }
+
+        return $this->emoji;
+    }
+
+    public function getException(): ?\Throwable
+    {
+        return $this->exception;
+    }
+
+    public function getExceptionAsString(): string
+    {
+        if (!$this->exception) {
+            return '';
+        }
+
+        if (class_exists(FlattenException::class)) {
+            $exception = $this->exception instanceof FlattenException ? $this->exception : FlattenException::createFromThrowable($this->exception);
+
+            return $exception->getAsString();
+        }
+
+        $message = \get_class($this->exception);
+        if ('' !== $this->exception->getMessage()) {
+            $message .= ': '.$this->exception->getMessage();
+        }
+
+        $message .= ' in '.$this->exception->getFile().':'.$this->exception->getLine()."\n";
+        $message .= "Stack trace:\n".$this->exception->getTraceAsString()."\n\n";
+
+        return rtrim($message);
+    }
+
+    /**
+     * @param array $channels An array of channels (entries are ANDed and sub-arrays are ORed)
```

Comment:
It looks like this is not implemented

Reply:
indeed, moved to the transport layer, fixed